tests/surveys/complex-survey/submissions.json

```diff
@@ -0,0 +1,38 @@
+{
+    "valid": [
+        {
+            "1": "noreply@fastsurvey.io",
+            "2": true,
+            "3": {
+                "1": false,
+                "2": false,
+                "3": false,
+                "4": true
+            },
+            "4": {
+                "1": true,
+                "2": false,
+                "3": true
+            },
+            "5": "It keeps my head cool while I answer tricky surveys."
+        }
+    ],
+    "invalid": [
+        {
+            "1": "noreply@fastsurvey.io",
+            "2": true,
+            "3": {
+                "1": false,
+                "2": false,
+                "3": false,
+                "4": true
+            },
+            "4": {
+                "1": 3.1415,
+                "2": false,
+                "3": true
+            },
+            "5": "It keeps my head cool while I answer tricky surveys."
+        }
+    ]
+}
```

Comment:
How about a few other invalid cases:

- One field is missing
- One Option is missing
- One Option/Field too many
- One Field in the wrong format (not just a single option), e.g. a string instead of a dict on field "4"